.framework/migration_optimizer.sh

```
@@ -0,0 +1,73 @@
+        fi
+
+        if [[ "$has_private_migrations" = false ]]; then
+            return 1  # No pending migrations
+        fi
+    fi
+
+    # Check cache file age (24 hours)
+    if [[ -f "$DF_MIGRATION_CACHE" ]]; then
+        local cache_age=$(($(date +%s) - $(stat -c %Y "$DF_MIGRATION_CACHE" 2>/dev/null || stat -f %m "$DF_MIGRATION_CACHE" 2>/dev/null || echo 0)))
+        if [[ $cache_age -lt 86400 ]]; then
+            local cache_result=$(cat "$DF_MIGRATION_CACHE" 2>/dev/null || echo "check")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)